src/dotty/tools/dotc/transform/Space.scala

```diff
@@ -14,35 +14,41 @@ import core.Decorators._
   */
 
 
-// space definition
+/** space definition */
 sealed trait Space
-object Empty extends Space
+case object Empty extends Space
 case class Typ(t: Type) extends Space
 case class Kon(t: Type, params: List[Space]) extends Space
 case class Or(spaces: List[Space]) extends Space
 
-// abstract space logic
+/** abstract space logic */
 trait SpaceLogic {
-  def isSubType(t1: Type, t2: Type): Boolean
-  def isEqualType(t1: Type, t2: Type): Boolean
-  def isCaseClass(t: Type): Boolean
+  /** is `tp1` a subtype of `tp2` */
+  def isSubType(tp1: Type, tp2: Type): Boolean
 
-  // Can a type be decompsed? i.e. all values of the type can be covered
-  // by its decomposed types.
-  //
-  // abstract sealed class and OrType can be decomposed
-  def canDecompose(t: Type): Boolean
+  /** is `tp1` the same type as `tp2` */
+  def isEqualType(tp1: Type, tp2: Type): Boolean
 
-  // return parameters types of case class
-  def getSignature(t: Type): List[Type]
+  /** is `tp` a case class */
+  def isCaseClass(tp: Type): Boolean
 
-  // get components of decomposable types
-  def getPartitions(t: Type): List[Type]
+  /** is the type `tp` decompsable? i.e. all values of the type can be covered
+    * by its decomposed types.
```

Comment:
It's weird but the comment style in dotty is different from the normal style, you should write:
```scala
/** Line 1
 *  Line 2
 *  Line 3
 */
Not:
```scala
/** Line 1
  * Line 2
  * Line 3
  */
```
See `SymDenotations` for an example of mostly well-documented code.